polar-core/tests/integration_tests.rs

```diff
@@ -1484,3 +1486,20 @@ fn test_external_unify() {
     let (results, _externals) = query_results_with_externals(query);
     assert!(results.is_empty());
 }
+
+#[test]
+fn test_list_results() {
```

Comment:
This is an _awesome_ test. But could we have a few simpler examples? This logic seems really hard to get right. How about:

- `[1, 2] = [1, *ys]` outputs `xs= [2]`
- `[1, 2, *xs] = [1, *ys] and [1, 2, 3] = [1, *ys]` outputs `xs = [3], ys = [2, 3]`
- `xs = [2], [1, 2] = [1, *xs]` succeeds
And then failure version of each of those.

Reply:
That second one in particular seems tough

Reply:
Those all pass (after I removed the typos). I don't know what you mean by "the failure version of each of those".

Reply:
So maybe something like `[1, 2] = [2, *ys]` should fail, `[1, 2, *xs] = [1, *ys] and [1, 2, *ys]` should fail.

Reply:
`[1, 2] = [2, *ys]` fails correctly.
`[1, 2, *xs] = [1, *ys] and [1, 2, *ys]` doesn't make sense.

Reply:
My bad, `[1, 2, *xs] = [1, *ys] and [1, 2, 3] = [1, 2, *ys]`